### PR TITLE
Adding contributor functionality

### LINKED ISSUE #2349: Support for multiple contributors
Currently the application allows for 1 PI and 1 "Data Contact Person." In order to support machine actionability of DMPs, we need to expand the number of people that can be linked to a DMP. This could be a repeatable field so that a user could add as many PIs and Data Curators as applicable. This may need a specific own tab in the application. We'd like to make the PI & data contact mandatory.

### PR DESCRIPTION
**Pull request by briri**
Fixes #2349.

- Add new `contributors` table with f-keys on Plan and Org
- Add Contributor model
- Add new routes, controllers, javascript and corresponding views
- Added tests

### CODE REVIEW

File: app/controllers/api/v0/plans_controller.rb [hunk 1]
```diff
@@ -23,17 +23,19 @@ def create
     # ensure user's organisation is the same as api user's
     unless plan_user.org == @user.org
       raise Pundit::NotAuthorizedError, _("user must be in your organisation")
-    end
+    ends
```
Comment (xsrust) [inline app/controllers/api/v0/plans_controller.rb:26]: think this might be a typo

File: app/controllers/api/v0/plans_controller.rb [hunk 2]
```diff
@@ -23,17 +23,19 @@ def create
...
 
     # initialize the plan
-    @plan = Plan.new
-    if plan_user.surname.blank?
-      @plan.principal_investigator = nil
-    else
-      @plan.principal_investigator = plan_user.anem(false)
-    end
+    @plan = Plan.news
```
Comment (xsrust) [inline app/controllers/api/v0/plans_controller.rb:29]: Think this one may be a typo as well

File: app/controllers/contributors_controller.rb [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    @contributor.identifiers = []
+  end
+
+  def save_orcid
+    return true unless @cached_orcid.present?
+
+    @cached_orcid.identifiable = @contributor
+    @cached_orcid.save
+    @contributor.reload
+  end
```
Comment (xsrust) [inline app/controllers/contributors_controller.rb:187]: Thanks for describing the workaround here.

File: app/controllers/plans_controller.rb [hunk 4]
```diff
@@ -253,7 +241,7 @@ def update
         #@plan.save
         if @plan.update(attrs) #_attributes(attrs)
           format.html do
-            redirect_to overview_plan_path(@plan),
+            redirect_to plan_contributors_path(@plan),
```
Comment (xsrust) [inline app/controllers/plans_controller.rb:244]: I like the idea of routing them to the contributors page after updating the plan

File: app/controllers/api/v0/plans_controller.rb (hunk 2 above)
Comment (briri) [reply app/controllers/api/v0/plans_controller.rb:29]: ugh yes. I'm surprised the tests passed!

**Comment by briri** [comment]
closing. will resubmit PR to development

**Pull request closed without merging**